plugins/inputs/prometheus/prometheus.go

```diff
@@ -427,6 +433,12 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 	return nil
 }
 
+func (p *Prometheus) addHeaders(req *http.Request) {
+	for header, value := range p.headers {
+		req.Header.Set(header, value)
```

Comment:
I was just copying some other addHeaders code. all good.

```suggestion
		req.Header.Add(header, value)
```